{{ cookiecutter.package_name }}/.travis.yml

```diff
@@ -60,6 +60,26 @@ env:
         # line to use the X virtual framebuffer.
         # - SETUP_XVFB=True
 
+        # PEP8 errors/warnings:
+        # E101 - mix of tabs and spaces
+        # W191 - use of tabs
+        # W291 - trailing whitespace
+        # W292 - no newline at end of file
+        # W293 - trailing whitespace
+        # W391 - blank line at end of file
+        # E111 - 4 spaces per indentation level
+        # E112 - 4 spaces per indentation level
+        # E113 - 4 spaces per indentation level
+        # E502 - the backslash is redundant between brackets
+        # E722 - do not use bare except
+        # E901 - SyntaxError or IndentationError
+        # E902 - IOError
+        # E999: SyntaxError -- failed to compile a file into an Abstract Syntax Tree
+        # # F821: undefined name  # Note: Removed for now because of heavy use of units.si
+        # F822: undefined name in __all__
+        # F823: local variable name referenced before assignment
+        - FLAKE8_OPT="--select=E101,W191,W291,W292,W293,W391,E111,E112,E113,E502,E722,E901,E902,E999,F822,F823"
```

Comment:
Since this is for new projects, why not go with the default of testing everything?

Reply:
That is, we could always default FLAKE8_OPT to ``''`` and then just have a commented out example of how to select specific errors, but I personally don't think we need the whole list here.

Reply:
Personally, I start with everything and then use `ignore` instead of `select` over at `setup.cfg`. That way, what Travis complains about is conistent with what Emacs complains about. However, there was a reason that Astropy core is not doing that. Not sure how you want to deal with the template.

Reply:
I agree with that approach - What about putting something like:

```
# If you want to ignore certain flake8 errors, you can list them in FLAKE8_OPT, for example:
# FLAKE8_OPT='--ignore=E501'
FLAKE8_OPT=''
```

Reply:
I'm 👍 for defaulting for testing everything. Astropy code doesn't do that as there was a pushback from a few maintainers, but for any new projects, especially with a smaller team can presumably write consistent code more easily from day1.

Also, I like adding ignores rather than selects very much.